File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TDBXAnsiStringValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * 
 * Wrap the AnsiString type and allows it to be null
 *
 */

public class TDBXAnsiStringValue extends DBXValue {
	protected boolean ValueNull = false;
	private String DBXStringValue;

	public TDBXAnsiStringValue() {
		super();
		setDBXType(DBXDataTypes.WideStringType);
	}
	
	public void setNull() {
		ValueNull = true;
		DBXStringValue = null;
	}

	public boolean isNull() {
		return ValueNull;
	}

	public void SetAsString(String Value) throws DBXException {
		ValueNull = false;
		DBXStringValue = Value;
	}

	public String GetAsString() throws DBXException {
		return DBXStringValue;
	}
}

File: Demos/Datasnap/Server/proxy/csharp_silverlight/TDBXAnsiStringValue.cs
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

using System;

namespace Embarcadero.Datasnap.WindowsPhone7
{
    /**
     * 
     * Wraps the AnsiString type and allows it to be null
     *
     */

    public class TDBXAnsiStringValue : DBXValue {
	    protected bool ValueNull = false;
	    private String DBXAnsiStringValue;

        /**
    	 * Class constructor, initialized this {@link DBXDataTypes} like a WideStringType 
    	 */
	    public TDBXAnsiStringValue() : base() {
		    setDBXType(DBXDataTypes.WideStringType);
	    }

         /**
    	 * Sets this object to null
    	 */
        public override void setNull()
        {
		    ValueNull = true;
		    DBXAnsiStringValue = "";
	    }

        /**
	     * Returns true if this object is null false otherwise.
	    */
        public override bool isNull()
        {
		    return ValueNull;
	    }

         /**
    	 * Sets the internal value with the value passed
    	 */
        public override void SetAsString(String Value)
        {
		    ValueNull = false;
		    DBXAnsiStringValue = Value;
	    }

        /**
    	 * Returns the internal value
    	 */
        public override String GetAsString()
        {
		    return DBXAnsiStringValue;
	    }
    }
}
